subprojects/packagefiles/cfitsio/meson.build

```diff
@@ -0,0 +1,270 @@
+project(
+  'cfitsio',
+  'c',
+  version: '4.5.0',
+  meson_version: '>=1.1.0',
+  license: 'CFITSIO',
+  license_files: ['licenses/License.txt'],
+)
+has_fortran = add_languages(
+  'fortran',
+  native: false,
+  required: get_option('fortran'),
+)
+
+SOVERSION = 10
+
+cc = meson.get_compiler('c')
+libm = cc.find_library('m')
+
+if target_machine.system() != 'windows'
+  curl = dependency(
+    'libcurl',
+    required: get_option('curl'),
+    fallback: ['curl', 'libcurl'],
+  )
+  if curl.found()
+    add_project_arguments('-DCFITSIO_HAVE_CURL', language: 'c')
+  endif
+endif
+
+threads = dependency('threads', required: get_option('reentrant'))
+if threads.found()
+  add_project_arguments('-D_REENTRANT', language: 'c')
+endif
+
+zlib = dependency('zlib', fallback: ['zlib', 'zlib_dep'])
+
+if target_machine.system() != 'windows'
+  bzip2 = dependency(
+    'bzip2',
+    required: get_option('bzip2'),
+    fallback: ['bzip2', 'bzip2'],
+  )
+  if bzip2.found()
+    add_project_arguments('-DHAVE_BZIP2=1', language: 'c')
+  endif
+endif
+
+if get_option('hera')
+  add_project_arguments('-DBUILD_HERA=1', language: 'c')
+endif
+
+if cc.get_id() != 'msvc' and cc.has_function('gethostbyname') and cc.has_function(
+  'connect',
+)
+  add_project_arguments('-DHAVE_NET_SERVICES', language: 'c')
+  if cc.has_header_symbol('stdio.h', 'fmemopen')
+    add_project_arguments('-DHAVE_FMEMOPEN', language: 'c')
+  endif
+endif
+
+if get_option('sse2')
+  if cc.has_argument('-msse2')
+    add_project_arguments('-msse2', language: 'c')
+  elif cc.get_id() == 'msvc'
+    add_project_arguments('-D__SSE2__=1')
+  endif
+endif
+
+if get_option('ssse3')
+  if cc.has_argument('-mssse3')
+    add_project_arguments('-mssse3', language: 'c')
+  elif cc.get_id() == 'msvc'
+    add_project_arguments('-D__SSE2__=1', '-D__SSSE3__=1', language: 'c')
+  endif
+endif
+
+if cc.has_header_symbol('unistd.h', 'ftruncate')
+  add_project_arguments('-DHAVE_UNISTD_H', '-DHAVE_FTRUNCATE', language: 'c')
+endif
+
+shmem_prefix = [
+  '#include <sys/ipc.h>',
+  '#include <sys/shm.h>',
+  '#include <sys/sem.h>',
+]
+flock_prefix = ['#include <sys/flock.h>']
+
+have_shmem_services = true
+foreach func : ['shmat', 'shmdt', 'shmget', 'semget']
+  have_shmem_services = have_shmem_services and cc.has_function(
+    func,
+    prefix: shmem_prefix,
+  )
+endforeach
+if have_shmem_services
+  add_project_arguments('-DHAVE_SHMEM_SERVICES', language: 'c')
+  if cc.has_type('flock_t', prefix: flock_prefix)
+    add_project_arguments('-DHAVE_FLOCK_T', language: 'c')
+  endif
+  if cc.has_type('union semun', prefix: shmem_prefix)
+    add_project_arguments('-DHAVE_UNION_SEMUN', language: 'c')
+  endif
+endif
+
+if cc.has_type('long long')
+  add_project_arguments('-DHAVE_LONGLONG', language: 'c')
+endif
+
+install_headers('fitsio.h', 'fitsio2.h', 'longnam.h')
+
+libcfitsio_la_SOURCES = [
```

Comment:
The `_la` refers to "libtool archive" so I wonder how useful it is to have that in the name.

Reply:
removed